src/__test__/components/ErrorBoundary.test.jsx

```diff
@@ -19,16 +20,28 @@ jest.mock('utils/postErrorToSlack');
 delete window.location;
 window.location = { reload: jest.fn() };
 
-const ThrowError = ({ hasError }) => {
-  if (hasError) throw new Error('test');
-  return content;
+const throwError = () => {
+  userEvent.click(screen.getByText('Throw'));
+};
+
+const TestComponent = () => {
+  const [hasError, setHasError] = useState(false);
+
+  if (hasError) throw new Error('Error');
+
+  return (
+    <>
+      <button type='button' onClick={() => setHasError(true)}>Throw</button>
```

Comment:
is there any reason to use this `setHasError(true)` instead of directly throwing the error like this:
`<button type='button' onClick={() => throw new Error('Error')}>Throw</button>`

Reply:
That's the difference between errors that can be caught by React error boundary and those that can't. As stated in [the docs](https://reactjs.org/docs/error-boundaries.html), React error boundary does not catch errors that originate in error handlers. It seems to only catch errors that originate during the generation of React views - i.e. errors arising in the body of functional components.